proxy/src/auth/backend/mod.rs

```diff
@@ -368,11 +261,13 @@ async fn auth_quirks(
     )
     .await
     {
-        Ok(keys) => Ok((keys, Some(allowed_ips.as_ref().clone()))),
+        Ok(keys) => Ok((keys, access_controls)),
         Err(e) => {
             if e.is_password_failed() {
                 // The password could have been changed, so we invalidate the cache.
-                cached_entry.invalidate();
+                // We should only invalidate the cache if the TTL might have expired.
+
+                // TODO: cached_entry.invalidate();
```

Comment:
Do we still need this TODO and comment?

Reply:
No, I moved the invalidation up the stack
https://github.com/neondatabase/neon/blob/d71792d6d6f7e5eb408ef1c61a9e30949b9122bf/proxy/src/auth/backend/mod.rs#L360-L371
Will remove.